controller/model/identity_manager.go

```diff
@@ -145,6 +146,74 @@ func (self *IdentityManager) ApplyCreateWithEnrollments(cmd *CreateIdentityWithE
 	})
 }
 
+func (self *IdentityManager) CreateWithAuthenticators(identity *Identity, authenticators []*Authenticator, ctx *change.Context) (string, []string, error) {
+	if identity.Id == "" {
+		identity.Id = eid.New()
+	}
+
+	var authenticatorIds []string
+	for _, authenticator := range authenticators {
```

Comment:
does this mean the callee can send in authenticators that have id's already? is that a common pattern? seems odd to me to allow the callee to specify the id?